docs/fastdds/getting_started/simple_app/includes/task_subscriber.rst

```diff
@@ -0,0 +1,118 @@
+the data reader class.
+
+*   Subscriber. Is the object responsible for the actual reception of the data.
+*   DataReader. It registers in the application the topic (TopicDescription) that identifies the data to be read and
+    accesses the data received by the subscriber.
+*   DataReaderListener. This is the listener assigned to the data reader.
+*   DataReaderQoS. Structure that defines the QoS of the DataReader.
+*   SampleInfo. Is the information that accompanies each sample that is ‘read’ or ‘taken.’
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 22,29
+
+The next line defines the HelloWorldSubscriber class that implements a subscriber.
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 33
+
+The public constructor and destructor of the class is defined below. The fields of the class will be the participant,
+the subscriber, the topic, the data reader, and the data type.
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 37-61
+
+
+As for the private fields of the class, it is worth mentioning the implementation of the DataReader listener.
+As it was the case with the DataWriter, the listener implements the callbacks to be executed in case an event
+occurs.
+The first overridden callback of the SubListener is the ``on_subscrition_matched``, which is the analog of the
+``on_publication_matched`` callback of the DataWriter.
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 140-157
+
+The second overridden callback is ``on_data_available``.
+In this, the next received sample that the DataReader can access is taken and processed to display its content.
+It is here that the object of the SampleInfo class is defined, which determines whether a sample has already been read
+or taken.
+Each time a sample is read, the counter of samples received is increased.
+
+.. literalinclude:: /../code/Examples/C++/DDSHelloWorld/src/HelloWorldSubscriber.cpp
+    :language: C++
+    :lines: 159-172
+
+Back to the public methods of the HelloWorldSubscriber class, we have the subscriber initialization method.
+This is the same as the initialization method defined for the publisher.
+However, in this one the DataReader QoS values are modified.
+First the reliability of the DataReader is modified to be Reliable, namely RELIABLE_RELIABILITY_QOS.
+This allows the reader to request the samples that have been lost in the transmission.
+Also, we modify the durability of the DataReader which is labeled as Transient-Local, namely
+TRANSIENT_LOCAL_DURABILITY_QOS.
+The DataReader will request all the samples that the publisher has sent before the subscriber has been matched
+with the publisher.
```

Comment:
As I said in the code, since the publisher is waiting for discovery, we don't need to "over-complicate" the example with these